File: objc/Command/Command/Book.m
#import "Book.h"

@implementation Book

-(void)buy {
    NSLog(@"Buying the book...");
    _currentOwner = @"Me";
}

-(void)sell {
    NSLog(@"Selling the book...");
    _currentOwner = @"Library";
}

-(void)giftToFriend:(NSString*)friendName {
    NSLog(@"Sending the book as gift to %@", friendName);
    _currentOwner = @"Friend";
}

@end

File: swift/Command/Command/Book.swift
import Foundation

public class Book {
    var currentOwner: String?
    
    func buy() -> Void {
        print("Buying the book...")
        currentOwner = "Me"
    }
    
    func sell() -> Void {
        print("Selling the book...")
        currentOwner = "Library"
    }
    
    func giftToFriend(_ friendName: String) -> Void {
        print("Sending the book as gift to \(friendName)")
        currentOwner = "Friend"
    }
}
